pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -320,6 +320,17 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 		return reconcile.Result{}, r.Update(context.TODO(), cd)
 	}
 
+	// Set platform label on the ClusterDeployment
+	platform := getClusterPlatform(cd)
+	if cd.Labels == nil {
+		cd.Labels = make(map[string]string)
+	}
+	cdPlatform, cdPlatformLabelExists := cd.Labels[hivev1.HiveClusterPlatformLabel]
+	if !cdPlatformLabelExists || cdPlatform != platform {
+		cd.Labels[hivev1.HiveClusterPlatformLabel] = platform
+		return reconcile.Result{}, r.Update(context.TODO(), cd)
```

Comment:
Please log a non-nil error returned from `Update`. Use `controllerutils.LogLevel(err)` as the log level of the log entry emitted.